Shoko.Server/Providers/TMDB/TmdbMetadataService.cs

```diff
@@ -2105,6 +2100,28 @@ private void PurgeCompany(int companyId, bool removeImageFiles = true)
             return (newlyAdded, updated);
         }
     }
+    
+    private async Task UpdateMoviesAndShowsByPerson(int tmdbPersonId)
+    {
+        var showIds = new HashSet<int>();
+        var movieIds = new HashSet<int>();
+
+        foreach (var staff in _tmdbEpisodeCast.GetByTmdbPersonID(tmdbPersonId))
+            showIds.Add(staff.TmdbShowID);
+        foreach (var staff in _tmdbEpisodeCrew.GetByTmdbPersonID(tmdbPersonId))
+            showIds.Add(staff.TmdbShowID);
+
+        foreach (var staff in _tmdbMovieCast.GetByTmdbPersonID(tmdbPersonId))
+            movieIds.Add(staff.TmdbMovieID);
+        foreach (var staff in _tmdbMovieCrew.GetByTmdbPersonID(tmdbPersonId))
+            movieIds.Add(staff.TmdbMovieID);
+
+        foreach (var showId in showIds)
+            await UpdateShow(showId, downloadCrewAndCast: true);
```

Comment:
If you schedule the updates to run in the queue, then you can just await the method and let the queue auto scale these in parallel (if possible).